bin/validate.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+
+import jsonschema
+from jsonschema import validate
+import yaml
+from yaml import load, dump, Loader, Dumper
+import click
+import os
+import sys
+
+script_dir = os.path.dirname(os.path.realpath(__file__))
+
+def load_doc(filename):
+    with open(filename, "r") as stream:
+    	try:
+        		return yaml.safe_load(stream)
+    	except yaml.YAMLError as exc:
+        		print(exc)
+
+def validate_config(config):
+    try:
+        validate(instance=config, schema=schema)
+    except jsonschema.exceptions.ValidationError as err:
+        print(err, file=sys.stderr)        
+        return False
+    return True
+
+def check_slot(adata, slot_type, slot_name):
```

Comment:
I would suggest some pytest/doctests for these methods, besides load_doc. I have some small AnnData for galaxy seurat testing that could be used, and they could be downloaded from elsewhere if needed.

Reply:
(they are sub selections that work with most methods on one of our smallest datasets).

Reply:
Yep, will add tests when I've finalised how things will work.

Reply:
(P.S. Scanpy has embedded test data I normally use for things like this)